pkg/syncer/util.go

```diff
@@ -482,6 +495,11 @@ func getVcHostFromTopologySegments(ctx context.Context, topologySegments []map[s
 	log := logger.GetLogger(ctx)
 	var vcHost string
 
+	if len(topologySegments) == 0 {
+		return "", logger.LogNewErrorf(log,
+			"Invalid volume %s as it does not have node affinity rules", volumeName)
```

Comment:
Is this message printed as event on the PV?
what about RWM PV. We do not want to publish affinity rules on RWM PV.

Reply:
> Is this message printed as event on the PV? what about RWM PV. We do not want to publish affinity rules on RWM PV.

We are not calling this method for RWX volumes as there are no PV affinity rules on it.